ui/component/viewers/videoViewer/internal/plugins/videojs-snapshot-button/plugin.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+import videojs from 'video.js';
+import type { Player } from '../../videojs';
+
+const VERSION = '1.0.0';
+const defaultOptions = {};
+
+type Options = {
+  fileTitle: string,
+};
+
+function onPlayerReady(player: Player, options: Options) {
+  // this doesn't work, but it should
+  // player.children()[0].setAttribute('crossorigin', 'anonymous');
+  const button = videojs.getComponent('Button');
+  const snapshotButton = videojs.extend(button, {
+    constructor: function () {
+      button.apply(this, arguments);
+      this.controlText('Take A Snapshot');
+      this.addClass('vjs-snapshot-button');
+      this.el_.innerHTML =
+        '<svg style="fill:#d1d1d1;height:25px;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path d="M149.1 64.8L138.7 96H64C28.7 96 0 124.7 0 160V416c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V160c0-35.3-28.7-64-64-64H373.3L362.9 64.8C356.4 45.2 338.1 32 317.4 32H194.6c-20.7 0-39 13.2-45.5 32.8zM256 384c-53 0-96-43-96-96s43-96 96-96s96 43 96 96s-43 96-96 96z"/></svg>';
+      this.el_.style.fill = 'white;';
+      this.el_.style.height = '20px;';
+    },
+    handleClick: function () {
+      const link = document.createElement('a');
+      const video = document.querySelector('video');
+
+      // $FlowIssue
+      const width = player.videoWidth();
+      // $FlowIssue
+      const height = player.videoHeight();
```

Comment:
Good point about defining in `Player`.

For the `drawImage` Flow is complaining that the number of arguments is invalid though it's a valid way to call the method.
![Screen Shot 2022-10-20 at 6 51 07 PM](https://user-images.githubusercontent.com/7200471/197011429-4bc0e1ef-f1e8-42f7-a3a3-730fc231ff7c.JPG)
![Screen Shot 2022-10-20 at 6 52 19 PM](https://user-images.githubusercontent.com/7200471/197011442-ed6da4e3-43ee-48ff-b2bd-0e04fbe7f317.JPG)

Reply:
I believe that is the wrong interpretation of the error, because the exact code in `ModalAutoGenerateThumbnail` had no issues.  The message is innacurate.

The actual issue should be because flow couldn't figure out what type `width` and `height` is, so it couldn't match the right overload for `drawImage`. Once you fix the initial problem (i.e. defined  Player's `videoWidth: () => number`), this warning should go away as well.

Reply:
Ah, it requires another `if (!video) return` to complete everything without suppression.  So, technically flow is still correct, whether or not that scenario will ever happen.

<sub>(when null or undefined)</sub>
<img width="375" alt="image" src="https://user-images.githubusercontent.com/64950861/197109977-a9569cb3-557d-4454-a301-932b7ddc6541.png">

----

It does bring up another point:  is `const video = document.querySelector('video');` guaranteed to always be pointing to our video?

I know Haffa had issues with Ads before:

```
    // avoid selecting 'video' anymore -> can cause conflicts with Ad popup videos
    const videoOnPage = document.querySelector('.vjs-tech');
```

There is a snapshot error in floating player mode, not sure if due to this or not:

<img width="777" alt="image" src="https://user-images.githubusercontent.com/64950861/197111308-aab02b2d-1626-4b05-a2e7-ac2febcbbfab.png">

Reply:
All should be addressed, only issue is that with:

This works, if it's run conditionally based on `video` existing, but with a more specific query selector it errors: 
      const video = document.querySelector('video');
      canvas.getContext('2d').drawImage(video, 0, 0, width, height);

Even if existence of video is conditionally checked for here, Flow will error.
      const video = document.querySelector('video.vjs-tech');
      canvas.getContext('2d').drawImage(video, 0, 0, width, height);